File: entities/src/main/java/net/kemitix/naolo/entities/Visit.java
package net.kemitix.naolo.entities;


import lombok.*;

import javax.json.bind.annotation.JsonbDateFormat;
import javax.persistence.*;
import java.time.LocalDateTime;

@EntityListeners({
        JPAActivityListener.class
})
@Entity
@NamedQuery(name = Visit.FIND_ALL,
        query = "select v from Visit v order by v.dateTime")
@With
@Getter
@ToString
@EqualsAndHashCode
@NoArgsConstructor(force = true)
@AllArgsConstructor
public class Visit
        implements HasId {

    public static final String FIND_ALL = "Visit.FindAll";

    @Id
    @GeneratedValue
    long id;

    @ManyToOne
    Pet pet;

    @ManyToOne
    Veterinarian veterinarian;

    @JsonbDateFormat("yyyy-MM-dd HH:mm")
    LocalDateTime dateTime;

    String description;
}

File: entities/src/main/scala/net/kemitix/naolo/entities/Visit.scala
package net.kemitix.naolo.entities

import java.time.ZonedDateTime

/**
  * A Visit by a Pet with a Veterinarian.
  */
object Visit {
  /**
    * Create a new Visit object.
    *
    * @param id          the Visit ID
    * @param petId       the Pet ID
    * @param vetId       the Veterinarian ID
    * @param dateTime    the Date and Time of the Visit
    * @param description the Description
    * @return a new Visit
    */
    def create(
                id: Long,
                petId: Long,
                vetId: Long,
                dateTime: ZonedDateTime,
                description: String
              ) =
      new Visit(id, petId, vetId, dateTime, description)
}

final class Visit private(
                           val id: Long,
                           val petId: Long,
                           val vetId: Long,
                           val dateTime: ZonedDateTime,
                           val description: String
                         ) {
  def getId: Long = id
  def getPetId: Long = petId
  def getVetId: Long = vetId
  def getDateTime: ZonedDateTime = dateTime
  def getDescription: String = description
}